### PR TITLE
fix: CYPRESS_DOWNLOAD_PATH_TEMPLATE placeholder

### LINKED ISSUE #23670: CYPRESS_DOWNLOAD_PATH_TEMPLATE only replace placeholder once
### Current behavior
```
CYPRESS_DOWNLOAD_PATH_TEMPLATE=http://tkeasydev.dst.tk-inline.net/nodejs/cypress/v${version}/cypress-${version}-${platform}-${arch}.zip

npm ERR! [FAILED] URL: http://tkeasydev.dst.tk-inline.net/nodejs/cypress/v10.7.0/cypress-${version}-win32-x64.zip
```

### Desired behavior
```
should replace all occurences of a value

CYPRESS_DOWNLOAD_PATH_TEMPLATE=http://tkeasydev.dst.tk-inline.net/nodejs/cypress/v${version}/cypress-${version}-${platform}-${arch}.zip

http://tkeasydev.dst.tk-inline.net/nodejs/cypress/v10.7.0/cypress-10.7.0-win32-x64.zip
```
### Test code to reproduce

no test available
failed on npm i

### Cypress Version

10.7.0

### Node version

18.8.0

### Operating System

Win 10

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by mahdikhashan**
- Closes #23670 

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by alex1701c** [comment]
@mahdikhashan Any update on this? I would really like this to get in :)

**Comment by mahdikhashan** [comment]
@alex1701c There have been an issue with it, other non-related tests are failing. Will give it a shot tomorrow in my spare time, hopefully will be fixed and people can merge it.

**Comment by alex1701c** [comment]
@mahdikhashan Any update?

**Comment by mahdikhashan** [comment]
@alex1701c it should be fixed now.

**Comment by alex1701c** [comment]
Thanks a lot! Let's hope this can get merged soonish :smile:

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ _Released 05/09/2023 (PENDING)_
 
 - Fixed an issue in Electron where devtools gets out of sync with the DOM occasionally. Addresses [#15932](https://github.com/cypress-io/cypress/issues/15932).
 - Updated the Chromium renderer process crash message to be more terse. Addressed in [#26597](https://github.com/cypress-io/cypress/pull/26597).
+- Fixed an issue in CYPRESS_DOWNLOAD_PATH_TEMPLATE regex. Addressed in [#23670](https://github.com/cypress-io/cypress/issues/23670).
```
Comment (mschile) [inline cli/CHANGELOG.md]:
```suggestion
- Fixed an issue with `CYPRESS_DOWNLOAD_PATH_TEMPLATE` regex to allow multiple replacements. Addresses [#23670](https://github.com/cypress-io/cypress/issues/23670).
```

File: cli/test/lib/tasks/download_spec.js [hunk 2]
```diff
@@ -81,6 +81,13 @@ describe('lib/tasks/download', function () {
       snapshot('desktop url from template with version', normalize(url))
     })
 
+    it('returns custom url from template with multiple replacements', () => {
+      process.env.CYPRESS_DOWNLOAD_PATH_TEMPLATE = '${endpoint}/v${version}/${platform}/${arch}/cypress-${version}-${platform}-${arch}.zip'
```
Comment (mschile) [inline cli/test/lib/tasks/download_spec.js]: Should probably test multiple endpoints as well:

```suggestion
      process.env.CYPRESS_DOWNLOAD_PATH_TEMPLATE = '${endpoint}/${platform}/${arch}/cypress-${version}-${platform}-${arch}.zip?referrer=${endpoint}&version=${version}'
```

**Review by ryanthemanuel — APPROVED** [review]

File: cli/__snapshots__/download_spec.js [hunk 3]
```diff
@@ -60,3 +60,7 @@ https://download.cypress.io/desktop/0.20.2/OS-ARCH/cypress.zip
 exports['desktop url from template with escaped dollar sign wrapped in quote'] = `
 https://download.cypress.io/desktop/0.20.2/OS-ARCH/cypress.zip
 `
+
+exports['desktop url from template with multiple variables'] = `
+https://download.cypress.io/desktop/0.20.2/v0.20.2/OS/ARCH/cypress-0.20.2-OS-ARCH.zip
```
Comment (mschile) [inline cli/__snapshots__/download_spec.js]:
```suggestion
https://download.cypress.io/desktop/0.20.2/OS/ARCH/cypress-0.20.2-OS-ARCH.zip?referrer=https://download.cypress.io/desktop/0.20.2&version=0.20.2
```

File: cli/test/lib/tasks/download_spec.js [hunk 4]
```diff
@@ -81,6 +81,13 @@ describe('lib/tasks/download', function () {
       snapshot('desktop url from template with version', normalize(url))
     })
 
+    it('returns custom url from template with multiple replacements', () => {
+      process.env.CYPRESS_DOWNLOAD_PATH_TEMPLATE = '${endpoint}/${platform}/${arch}/cypress-${version}-${platform}-${arch}.zip?referrer=${endpoint}&version=${version}'
+      const url = download.getUrl('ARCH', '0.20.2')
+
+      snapshot('desktop url from template with multiple variables', normalize(url))
```
Comment (mschile) [inline cli/test/lib/tasks/download_spec.js]:
```suggestion
      snapshot('desktop url from template with multiple replacements', normalize(url))
```

File: cli/__snapshots__/download_spec.js [hunk 5]
```diff
@@ -60,3 +60,7 @@ https://download.cypress.io/desktop/0.20.2/OS-ARCH/cypress.zip
 exports['desktop url from template with escaped dollar sign wrapped in quote'] = `
 https://download.cypress.io/desktop/0.20.2/OS-ARCH/cypress.zip
 `
+
+exports['desktop url from template with multiple variables'] = `
```
Comment (mschile) [inline cli/__snapshots__/download_spec.js]:
```suggestion
exports['desktop url from template with multiple replacements'] = `
```

**Review by mschile — APPROVED** [review]

**Comment by mschile** [comment]
Thanks for the contribution @mahdikhashan 🎉

**Pull request merged**